tests/test_integration.py

```diff
@@ -148,17 +148,17 @@ def setUp(self):
 
     def tearDown(self):
         """Clean up all HVs after every test"""
+
+        vm_obj = Query({'hostname': VM_HOSTNAME}, ['hostname']).get()
+        uid_name = '{}_{}'.format(vm_obj['object_id'], vm_obj['hostname'])
+
         for hv in HYPERVISORS:
             hv.run(
                 'virsh destroy {vm}; '
-                'virsh undefine {vm}'
-                .format(vm=VM_HOSTNAME),
-                warn_only=True,
-            )
-            hv.run(
+                'virsh undefine {vm}; '
                 'umount /dev/xen-data/{vm}; '
```

Comment:
While we're at it, `xen-data` should not be hardcoded but taken from settings.py